src/lib/components/ConfirmTransactions.js

```diff
@@ -12,78 +14,189 @@ ${json}
+
+  const getWidgetContractPermission = async (widgetSrc, contractId) =>
+    await cache.asyncLocalStorageGet(StorageDomain, {
+      widgetSrc,
+      contractId,
+      type: StorageType.SendTransactionWithoutConfirmation,
+    });
+
+  useEffect(() => {
+    (async () => {
+
+      if (transactions.length == 1) {
+        const contractId = transactions[0].contractName;
+        const isSignedIntoContract = await near.isSignedIntoContract(contractId);
+
+        const widgetContractPermission = await getWidgetContractPermission(widgetSrc, contractId);
+
+        const dontAskForConfirmation = (isSignedIntoContract && widgetContractPermission && widgetContractPermission[transactions[0].methodName])
+          ? true : false;
+
+        setDontAskForConfirmation(dontAskForConfirmation);
+
+        if (dontAskForConfirmation) {
+          setLoading(true);
+          const result = await near.sendTransactions(transactions);
+          setLoading(false);
+          onHide(result);
+        }
+      }
+    })();
+  }, []);
+
   const onHide = props.onHide;
-  const transactions = props.transactions;
+
   const show = !!transactions;
 
-  return (
-    <Modal size="xl" centered scrollable show={show} onHide={onHide}>
-      <Modal.Header closeButton>
-        <Modal.Title>Confirm Transaction</Modal.Title>
+  const dontAskAgainCheckboxChange = async () => {
+    setDontAskAgainChecked(!dontAskAgainChecked);
+  };
+
+  if (dontAskForConfirmation) {
+    const transaction = transactions[0];
+    return (<Modal size="xs" centered show={show} onHide={onHide}>
+      <Modal.Header>
+        <Modal.Title>Sending transaction</Modal.Title>
       </Modal.Header>
       <Modal.Body>
-        {transactions &&
-          transactions.map((transaction, i) => (
-            <div key={`${gkey}-${i}`}>
-              <div>
-                <h4>Transaction #{i + 1}</h4>
-              </div>
-              <div>
-                <span className="text-secondary">Contract ID: </span>
-                <span className="font-monospace">
-                  {transaction.contractName}
-                </span>
-              </div>
-              <div>
-                <span className="text-secondary">Method name: </span>
-                <span className="font-monospace">{transaction.methodName}</span>
-              </div>
-              {transaction.deposit && transaction.deposit.gt(0) && (
+        <div>
+          <span className="text-secondary">Contract name: </span>
+          <span className="font-monospace">{transaction.contractName}</span>
+        </div>
+        <div>
+          <span className="text-secondary">Method name: </span>
+          <span className="font-monospace">{transaction.methodName}</span>
+        </div>
+        <Markdown text={jsonMarkdown(transaction.args)} />
+      </Modal.Body>
+      <Modal.Footer>
+        {Loading}
+      </Modal.Footer>
+    </Modal>);
+  } else if (dontAskForConfirmation == false) {
```

Comment:
I guess this is enough:

```suggestion
  } else {
```

Reply:
`dontAskForConfirmation` is set by an async operation querying that the user is signed in and checking the widgets permission to call the contract without confirmation. The default state of this is `null`, while `false` confirms that we can show the confirmation popup.

Reply:
@petersalomonsen This is hard to maintain it this way. I suggest you handle the `null` case first, and then handle only true/false (if/else) branches